brouter-core/src/main/java/btools/router/OsmPath.java

```diff
@@ -346,8 +346,9 @@ protected void addAddionalPenalty(OsmTrack refTrack, boolean detailMode, OsmPath
         if ( rc.startDirectionValid )
         {
           double dir = rc.startDirection.intValue() / CheapRulerSingleton.DEG_TO_RAD;
-          lon0 = lon1 - (int) ( 1000. * Math.sin( dir ) / rc.getCosLat() );
-          lat0 = lat1 - (int) ( 1000. * Math.cos( dir ) );
+          double[] lonlat2m = CheapRulerSingleton.getLonLatToMeterScales( lat1 );
+          lon0 = lon1 - (int) ( 1000. * Math.sin( dir ) / lonlat2m[0] );
+          lat0 = lat1 - (int) ( 1000. * Math.cos( dir ) / lonlat2m[1] );

```

Comment:
`rc.coslat` is never updated anymore, so I simply discarded it. The only remaining use case was here, where I think we can use `lonlat2m` this way, but should be checked.